roachtest: clean up `failover` tests

erikgrinaker opened the pull request:
**roachtest: use `Logger.Printf()` instead of Test.Status() in `failover`**

**roachtest: don't log zone configs in `failover`**

**roachtest: set `failover` lease type on test spec**

**roachtest: don't disable load-based splits in `failover`**

There's no strong reason to disable load-based splits here, and using default settings makes for more realistic failures.

**roachtest: set up monitor early in `failover`**

This patch sets up the cluster monitor immediately after starting cluster nodes, to make the monitored node set more visible.

**roachtest: improve `failover` workload runner**

This patch changes the `failover` tests to run the workload until the test is done (context is cancelled), instead of using a fixed workload interval and trying to fit the failures/recoveries into that interval. This significantly simplifies the fail/recovery loops.

The workload still outputs performance artifacts when cancelled, which is what's needed for roachperf graphs.

**roachtest: improve failover test descriptions**

Epic: none
Release note: None

erikgrinaker committed:
roachtest: use `Logger.Printf()` instead of `Test.Status()` in `failover`

Epic: none
Release note: None

erikgrinaker committed:
roachtest: don't log zone configs in `failover`

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/103264)
<!-- Reviewable:end -->

erikgrinaker commented:
Note to self: address @andrewbaptist's comments in https://github.com/cockroachdb/cockroach/pull/103254#pullrequestreview-1429495288.

erikgrinaker commented:
> Wouldn't it be better to submit all the ranges to the replicate queue immediately? That would prevent any divergence between the ranges and the configuration above. (In this and other places). I think it should be similar speed to this approach, although it might still be necessary to do it in a loop. It would be much cleaner if this test just called `relocateRangesUntilStable` with no real parameters passed to it, and it continues running until there is nothing moved.

@andrewbaptist Do you have any suggestions for how we'd implement this? The only mechanism I can think of is `crdb_internal.kv_enqueue_replica()`, but that doesn't report anything about whether the replicate queue did anything with the range, only that it was successfully enqueued. It's also a bit iffy in that it errors out if there's no local leaseholder, so it'll be racy with concurrent lease/replica changes. We could speed up the scanner and replicate queue to process ranges more quickly, and then use replication reports to check whether ranges conform to zone constraints, but I believe replication reports are deprecated and about to be removed so I'm hesitant to build on top of them. Open to ideas for a canonical way of doing this, since it keeps coming up without any good options afaik.

erikgrinaker committed:
roachtest: don't disable load-based splits in `failover`

There's no strong reason to disable load-based splits here, and using
default settings makes for more realistic failures.

Epic: none
Release note: None

erikgrinaker committed:
roachtest: set up monitor early in `failover`

This patch sets up the cluster monitor immediately after starting
cluster nodes, to make the monitored node set more visible.

Epic: none
Release note: None

erikgrinaker committed:
roachtest: improve `failover` workload runner

This patch changes the `failover` tests to run the workload until the
test is done (context is cancelled), instead of using a fixed workload
interval and trying to fit the failures/recoveries into that interval.
This significantly simplifies the fail/recovery loops.

The workload still outputs performance artifacts when cancelled, which
is what's needed for roachperf graphs.

Epic: none
Release note: None

erikgrinaker committed:
roachtest: improve `failover` test descriptions

Epic: none
Release note: None

renatolabs commented on pkg/cmd/roachtest/tests/failover.go:
```diff
@@ -22,11 +23,43 @@ import (
 
+var rangeLeaseRenewalDuration time.Duration
+
+func init() {
```
Can we use the "func initializer" pattern (name I just made up, maybe there's a more official term), instead of a package-level `init`:

```go
var rangeLeaseRenewalDuration = func() time.Duration {
  // ...
}()
```

erikgrinaker replied:
Good idea, thanks. Done.

erikgrinaker committed:
roachtest: set `failover` lease type on test spec

srosenberg reviewed (commented):
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103264)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andrewbaptist, @erikgrinaker, @herkolategan, and @renatolabs)

___
*[`pkg/sql/sem/builtins/builtins.go` line 6317 at r15](https://reviewable.io/reviews/cockroachdb/cockroach/103264#-NVkJs4l4htghVWPPvo1:-NVkJs4l4htghVWPPvo2:b-ahr8em) ([raw file](https://github.com/cockroachdb/cockroach/blob/da89ba46474f2a6c6f8d5e204c8ec2ba2a31d52d/pkg/sql/sem/builtins/builtins.go#L6317)):*
> ```Go
> 				} else {
> 					// Unlocking a non-locked mutex will irrecoverably fatal the process.
> 					// We do a TryLock() to ensure the mutex is actually locked before
> ```

There is no guarantee that `TryLock` would succeed, right? If it fails, you could still end up calling `Unlock` when the mutex is not actually in the `mutexLocked` state, i.e., fatal the process. 


<!-- Sent from Reviewable.io -->

erikgrinaker reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103264)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andrewbaptist, @herkolategan, @renatolabs, and @srosenberg)

___
*[`pkg/sql/sem/builtins/builtins.go` line 6317 at r15](https://reviewable.io/reviews/cockroachdb/cockroach/103264#-NVkJs4l4htghVWPPvo1:-NVkySc-5aWS2_ovu8jt:b-nt9s4a) ([raw file](https://github.com/cockroachdb/cockroach/blob/da89ba46474f2a6c6f8d5e204c8ec2ba2a31d52d/pkg/sql/sem/builtins/builtins.go#L6317)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

There is no guarantee that `TryLock` would succeed, right? If it fails, you could still end up calling `Unlock` when the mutex is not actually in the `mutexLocked` state, i.e., fatal the process. 
</blockquote></details>

Correct, as the comment says this is racey and best-effort. This function is only for test use -- the caller is expected to already have locked the mutex, the `TryLock()` is just a cheap anti-footgun measure. The alternative is to not even try and always fatal.

In any case, let's discuss this change on the actual PR, #103261.


<!-- Sent from Reviewable.io -->

erikgrinaker commented:
Allright @andrewbaptist, this one's ready to go. If there's any other tweaks or cleanups you'd like to see then here's your chance, going to put the `failover` stuff aside after this and the renames in #103264.

andrewbaptist reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

Minor comment is that we should maybe move all these tests to weekly. The count is getting high, and since they are all passing and I assume they will going forward, it would be 99% as good at 1/7 the cost. Maybe let this commit go through for a couple days and then change it to weekly?

> Reviewed 1 of 1 files at r30, 1 of 1 files at r31, 1 of 1 files at r32, 1 of 1 files at r33, 1 of 1 files at r34, 1 of 1 files at r35, 1 of 1 files at r36, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103264#-:-NXG8dBI7RHV2dm03tRG:b-jhfpus)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @herkolategan, @renatolabs, and @srosenberg)



<!-- Sent from Reviewable.io -->

erikgrinaker commented:
bors r+

erikgrinaker commented:
TFTR!

bors r+
